forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,404 @@
+    pub chain_status: ColoredString,
+    pub network: ColoredString,
+    pub uptime: ColoredString,
+    pub health: ColoredString,
+    pub wallet_address: ColoredString,
+    pub wallet_balance: ColoredString,
+}
+
+#[derive(Debug, strum::Display, PartialEq)]
+pub enum SyncStatus {
+    Ok,
+    Slow,
+    Behind,
+}
+
+// Limits the format string from humantime::format_duration to exclude us ns ms parts.
+pub fn fmt_duration(duration: Duration) -> String {
+    let duration = humantime::format_duration(duration);
+    let duration = duration.to_string();
+    let duration = duration.split(' ');
+    let format_duration = duration
+        .filter(|s| !s.ends_with("us"))
+        .filter(|s| !s.ends_with("ns"))
+        .filter(|s| !s.ends_with("ms"))
+        .map(|s| s.to_string());
+    let format_duration: Vec<String> = format_duration.collect();
+    format_duration.join(" ")
+}
+
+impl NodeStatusInfo {
+    pub fn new(
+        cur_duration: Duration,
+        blocks_per_tipset_last_finality: f64,
+        head: TipsetJson,
+    ) -> anyhow::Result<NodeStatusInfo> {
+        let ts = head.0.min_timestamp();
+        let cur_duration_secs = cur_duration.as_secs();
+        let behind = if ts <= cur_duration_secs + 1 {
+            cur_duration_secs.saturating_sub(ts)
+        } else {
+            anyhow::bail!(
+                "System time should not be behind tipset timestamp, please sync the system clock."
+            );
+        };
+
+        let sync_status = if behind < EPOCH_DURATION_SECONDS as u64 * 3 / 2 {
+            // within 1.5 epochs
+            SyncStatus::Ok
+        } else if behind < EPOCH_DURATION_SECONDS as u64 * 5 {
+            // within 5 epochs
+            SyncStatus::Slow
+        } else {
+            SyncStatus::Behind
+        };
+
+        let base_fee = head.0.min_ticket_block().parent_base_fee().clone();
+
+        let health = 100. * blocks_per_tipset_last_finality / BLOCKS_PER_EPOCH as f64;
+
+        Ok(Self {
+            behind: Duration::from_secs(behind),
+            health,
+            epoch: head.0.epoch(),
+            base_fee,
+            sync_status,
+            start_time: Utc::now(),
+            network: String::from("unknown"),
+            default_wallet_address: None,
+            default_wallet_address_balance: None,
+        })
+    }
+
+    pub fn chain_status(&self) -> String {
+        let base_fee_fmt = self.base_fee.pretty();
+        let behind = format!("{}", humantime::format_duration(self.behind));
+        format!(
+            "[sync: {}! ({} behind)] [basefee: {base_fee_fmt}] [epoch: {}]",
+            self.sync_status, behind, self.epoch
+        )
+    }
+}
+
+impl From<NodeStatusInfo> for NodeInfoOutput {
+    fn from(node_status_info: NodeStatusInfo) -> NodeInfoOutput {
+        let health = node_status_info.health;
+        let uptime = (Utc::now() - node_status_info.start_time)
+            .to_std()
+            .expect("failed converting to std duration");
```

Comment:
`From` impls should *not* be nondeterministic

Reply:
Removed